### PR TITLE
[CG] PAC yang files update

### PR DESCRIPTION
**Pull request by chiranjeevi-u**
##### Work item tracking
- Microsoft ADO **(number only)**:

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-yang-models/yang-models/sonic-hostapd.yang [hunk 1]
```diff
@@ -6,10 +6,22 @@ module sonic-hostapd {
     description
         "SONiC HOSTAPD";
 
+    revision 2023-09-15 {
+        description "Updated with CG changes.";
```
Comment (thovikeerthi) [inline src/sonic-yang-models/yang-models/sonic-hostapd.yang:10]: Add description like, Update with LocalAuthentcation support for PAC changes, contributed from CG.

File: src/sonic-yang-models/yang-models/sonic-pac.yang [hunk 2]
```diff
@@ -3,13 +3,29 @@ module sonic-pac {
...
 
+    import sonic-interface {
+        prefix interface;
+    }
+
     description
         "SONiC PAC";
 
+    revision 2023-09-15 {
+        description "Updated with CG changes.";
```
Comment (thovikeerthi) [inline src/sonic-yang-models/yang-models/sonic-pac.yang:26]: Same as previous comment

File: src/sonic-yang-models/yang-models/sonic-pac.yang [hunk 3]
```diff
@@ -114,6 +130,25 @@ module sonic-pac {
         }
     }
 
+    typedef access_type_enumeration {
+        type enumeration {
+            enum allow;
+            enum deny;
```
Comment (thovikeerthi) [inline src/sonic-yang-models/yang-models/sonic-pac.yang:136]: As part of enumeration, add only 'allow' option. Put, deny in comment mode.

File: src/sonic-yang-models/yang-models/sonic-pac.yang [hunk 4]
```diff
@@ -114,6 +130,25 @@ module sonic-pac {
...
+        }
+    }
+
+    typedef user_auth_enumeration {
```
Comment (thovikeerthi) [inline src/sonic-yang-models/yang-models/sonic-pac.yang:140]: Please rename it as 'pac_auth_enumeration'

File: src/sonic-yang-models/yang-models/sonic-pac.yang [hunk 5]
```diff
@@ -230,5 +265,48 @@ module sonic-pac {
...
+
+            container global {
+                description "Global PAC configuration.";
+
+                leaf auth_order_list {
+                    type user_auth_enumeration;
+                    description "Order list of authentication mechanisms used by PAC. Must be 'remote' or 'local'.";
+                }
+            }
+        }
```
Comment (thovikeerthi) [inline src/sonic-yang-models/yang-models/sonic-pac.yang:310]: As per our previous conversation, please check the naming convention of the CLI commands getting generated from these 2 yang files and get the approval from Kishor.

**Review by thovikeerthi — COMMENTED** [review]
Please work on the comments and update back.

File: src/sonic-yang-models/yang-models/sonic-hostapd.yang (hunk 1 above)
Comment (chiranjeevi-u) [reply src/sonic-yang-models/yang-models/sonic-hostapd.yang]: Updated the description.

File: src/sonic-yang-models/yang-models/sonic-pac.yang (hunk 3 above)
Comment (chiranjeevi-u) [reply src/sonic-yang-models/yang-models/sonic-pac.yang:136]: Commented out the "enum deny".

File: src/sonic-yang-models/yang-models/sonic-pac.yang (hunk 4 above)
Comment (chiranjeevi-u) [reply src/sonic-yang-models/yang-models/sonic-pac.yang]: Renamed the typedef as per the comment received.

File: src/sonic-yang-models/yang-models/sonic-pac.yang [hunk 6]
```diff
@@ -3,13 +3,29 @@ module sonic-pac {
...
+
+    import sonic-types {
+        prefix stypes;
+    }
+
     import sonic-port {
         prefix prt;
     }
 
+    import sonic-interface {
```
Comment (thovikeerthi) [inline src/sonic-yang-models/yang-models/sonic-pac.yang:18]: This type is not used in current scope, please recheck if this update needed in the yang file.

File: src/sonic-yang-models/yang-models/sonic-pac.yang [hunk 7]
```diff
@@ -3,13 +3,29 @@ module sonic-pac {
     prefix spac;
     yang-version 1.1;
 
+    import ietf-yang-types {
+        prefix yang;
+    }
+
+    import sonic-types {
```
Comment (thovikeerthi) [inline src/sonic-yang-models/yang-models/sonic-pac.yang:10]: This type is not used in current scope, please recheck if this update needed in the yang file.

**Review by thovikeerthi — COMMENTED** [review]
Please check the comments and update back.

**Comment by thovikeerthi** [comment]
Approved

**Pull request merged**